src/main/java/org/ohdsi/circe/cohortdefinition/builders/BuilderUtils.java

```diff
@@ -136,9 +136,17 @@ public static String buildTextFilterClause(String sqlExpression, TextFilter filt
 
         String negation = filter.op.startsWith("!") ? "not" : "";
         String prefix = filter.op.endsWith("endsWith") || filter.op.endsWith("contains") ? "%" : "";
-        String value = filter.text;
         String postfix = filter.op.endsWith("startsWith") || filter.op.endsWith("contains") ? "%" : "";
 
+        String value = escapeSqlParam(filter.text);
+
         return String.format("%s %s like '%s%s%s'", sqlExpression, negation, prefix, value, postfix);
     }
+
+    private static String escapeSqlParam(String value) {
+        if (StringUtils.isEmpty(value)) {
+            return value;
+        }
+        return value.replaceAll("'", "''");
```

Comment:
In your unit test, you said that 'Simply doubling quotes doesn’t work', but looking at this code, it looks like you are just replacing single quotes with double quotes.   And, i don't see where this code applies the same logic as your test.

Can you confirm that you're fixing the sql injection?

Reply:
@chrisknoll you are absolutely right! This is probably a bad excuse, but I did not commit the latest changes. anyway, this case is covered and unit test is green.